Terms/SolrRegexTermsService.php

```diff
@@ -24,10 +24,9 @@ public function fetchTerms(SimpleQueryInterface $query)
             return new EmptyTermsResult();
         }
 
-        $suggestQuery
-            ->setFields($field)
-            ->setRegex($query->getSearchTerm())
-            ->setSort('count');
+        $suggestQuery->setFields($field);
+        $suggestQuery->setRegex($this->escapeTerm($query->getSearchTerm()));
```

Comment:
Yeah, doesn't work as in the query fails. Otherwise I'd have just used the helper method.